components/launcher-client/src/client.rs

```diff
@@ -124,40 +125,33 @@ impl LauncherCli {
 
     /// Send a process spawn command to the connected Launcher
     ///
-    /// `user` and `group` are string names, while `user_id` and
-    /// `group_id` are numeric IDs. Newer versions of the Launcher can
+    /// `username` and `groupname` are string names, while `uid` and
+    /// `gid` are numeric IDs. Newer versions of the Launcher can
     /// accept either, but prefer numeric IDs.
-    pub fn spawn<I, B, U, G, P>(&self,
-                                id: &I,
-                                bin: B,
-                                user: Option<U>,
-                                group: Option<G>,
-                                user_id: Option<u32>,
-                                group_id: Option<u32>,
-                                password: Option<P>,
-                                env: Env)
-                                -> Result<Pid>
-        where I: ToString,
-              B: AsRef<Path>,
-              U: ToString,
-              G: ToString,
-              P: ToString
-    {
```

Comment:
This is a pretty dramatic improvement for readability.

Reply:
@raskchanky What don't you get about I B U G P? :trollface:

Reply:
Heh, I mean I get the argument for using generics.  But it also rings true that if the same type is being passed at every call site, you aren't actually benefiting from the use of generics, and the code ends up being harder to read for no reason.

Reply:
Yeah, totally... I was just messin' 😄